### PR TITLE
Delete `GT_INDEX`

### PR DESCRIPTION
**Pull request by SingleAccretion**
Instead:
1\) For `ldelem`, import `IND/OBJ(INDEX_ADDR)`.
2\) For `ldelema`, import `INDEX_ADDR`.

This deletes two usages of `ADDR`:
1\) `OBJ(ADDR(INDEX))` from `ldelem<struct>`.
2\) `ADDR(INDEX)` from `ldelema`.

It also deletes a case of us creating layout-less `IND struct` nodes and some amount of auxiliary code.

[Diffs](https://dev.azure.com/dnceng/public/_build/results?buildId=1794970&view=ms.vss-build-web.run-extensions-tab): a few in the tests reflecting forward substitution's limit of the amount of nodes it is willing to forward, plus one in libraries tests stemming from [this](https://github.com/dotnet/runtime/blob/f8229bbe7f6642367241ed1158a912d169cc1252/src/coreclr/jit/gentree.cpp#L13356) code only looking at `ADDR`s (fix in progress: #69961). There is also a fairly small TP regression (curiously, 64 bit-only) that has been payed for with the improvements from #69895.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
Suspect the `Libraries Test Run checked coreclr Linux arm64 Debug` failure is #69657.

@dotnet/jit-contrib

File: src/coreclr/jit/gtlist.h [hunk 1]
```diff
@@ -147,9 +147,7 @@ GTNODE(COMMA            , GenTreeOp          ,0,GTK_BINOP|DBK_NOTLIR)
 GTNODE(QMARK            , GenTreeQmark       ,0,GTK_BINOP|GTK_EXOP|DBK_NOTLIR)
 GTNODE(COLON            , GenTreeColon       ,0,GTK_BINOP|DBK_NOTLIR)
 
-GTNODE(INDEX            , GenTreeIndex       ,0,GTK_BINOP|GTK_EXOP|DBK_NOTLIR) // SZ-array-element.
```
Comment (am11) [inline src/coreclr/jit/gtlist.h:150]: `GT_INDEX` is also mentioned in `docs/design/coreclr/jit/{first-class-structs.md,ryujit-tutorial.md}`.

Comment (SingleAccretion) [reply src/coreclr/jit/gtlist.h:150]: Thank you for pointing that out!

I deleted some obsolete contents from `first-class-structs.md`, `ryujit-tutorial.md` required more extensive changes so I opened #69988 and will do it separately.

**Review by BruceForstall — APPROVED** [review]
This looks great. I kicked off jitstress to verify with a larger test set.

**Comment by SingleAccretion** [comment]
[Stress was clean](https://dev.azure.com/dnceng/public/_build/results?buildId=1810215&view=results).

Force-pushed to resolve a conflict with #70265.

**Pull request merged**